corehq/apps/sms/views.py

```diff
@@ -1915,6 +1918,30 @@ def dispatch(self, request, *args, **kwargs):
         return super(SMSSettingsView, self).dispatch(request, *args, **kwargs)
 
 
+@method_decorator(toggles.ICDS_CUSTOM_SMS_REPORT.required_decorator(), name='dispatch')
+class SMSUsageReport(BaseMessagingSectionView):
+    template_name = 'sms/usage_report.html'
+    urlname = 'sms_usage_report'
+    page_title = _('SMS Usage Report')
+
+    @use_daterangepicker
+    def dispatch(self, *args, **kwargs):
+        return super(SMSUsageReport, self).dispatch(*args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+        try:
+            start_date, end_date = request.POST.get('date_range').split(' to ')
+            from corehq.messaging.scheduling.forms import validate_date
+            validate_date(start_date)
+            validate_date(end_date)
+            send_custom_sms_report.delay(start_date, end_date)
```

Comment:
Suggestion: you might consider limiting how wide a date range is allowed, since this report takes so long to run.

Reply:
Yep. Thanks for the suggestion, it really should be done.
But I have left it as of now, would wait for Priyanka's reply on the ticket then will add this functionality.